### PR TITLE
boards/mcb2388: add example application

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Often times having an example to show how certain APIs are used is the best documentation.
Some boards come with a rich set of peripherals that can not be covered by a generic example application.

This adds a board-specific example for the `mcb2388` to showcase the features of the board:
The 

 - HD44780 LCD display
 - Potentiometer and button as input
 - 8 debug LEDs mapped to a single port

### Testing procedure

The demo application implements a simple clock.

You can set the date and time using the potentiometer and button, then marvel at the ticking time.
The seconds are also displayed as binary on the debug LEDs.

### Issues/PRs references

with #13908 we can also play sound through the on-board speaker.
I already have patches for that.

### CODE REVIEW

File: makefiles/app_dirs.inc.mk [hunk 1]
```diff
@@ -9,6 +9,7 @@ RIOTBASE ?= .
 # 4. sort
 APPLICATION_DIRS := $(sort $(patsubst ./%,%,$(patsubst %/,%,$(dir $(wildcard \
 	$(RIOTBASE)/fuzzing/*/Makefile     \
+	$(RIOTBASE)/boards/*/examples/*/Makefile \
```
Comment (aabadie) [inline makefiles/app_dirs.inc.mk:12]: I would not put example applications in the boards directory. There are already the `examples` and tests` directories for that.

Comment (benpicco) [reply makefiles/app_dirs.inc.mk:12]: But that example is specific to the board.
That was a bit my intention with this PR - introduce board example applications that showcase some more complex applications to demo the features of a board.

E.g. we could also have one for `stm32f429i-disco` to showcase use of the touch screen/display/gyro that is on the board.
Sure we have tests for the individual components, but it's also good to have an example of how it all comes together.

Comment (aabadie) [reply makefiles/app_dirs.inc.mk:12]: Let's not add this here for the moment. There are already `tests/board_microbit` and `tests/board_calliope-mini` with the same purpose.
But I agree that they are a a bit hidden in the tests. Anyway this should be part of a more general discussion and rework of how applications are organized in RIOT.

**Comment by maribu** [comment]
Interesting: It seems the stale bot caused a Murdock run.

I agree with @aabadie that inside the board directory is not the right place for apps. And also that `tests` is not the place you usually expect examples in. How about `examples/board-showcases/<NAME_OF_BOARD>`?

**Comment by maribu** [comment]
Wasn't there some effort underway to bring more structure into the examples? So that we could add e.g. `examples/boards/<BOARD>` applications without having to risk that examples become impossible to navigate?

From the feedback I get from students, more example apps are something they would like to see.

**Pull request closed without merging**